frontend/src/lib/k8s/api/v2/hooks.ts

```diff
@@ -275,3 +287,122 @@ export function useKubeObjectList<T extends KubeObjectClass>({
     },
   };
 }
+
+/**
+ * Returns a combined list of Kubernetes objects and watches for changes from the clusters given.
+ */
+export function useKubeObjectList<T extends KubeObjectClass>({
+  kubeObjectClass,
+  namespace,
+  cluster,
+  clusters,
+  queryParams,
+}: {
+  /** Class to instantiate the object with */
+  kubeObjectClass: T;
+  /** Object list namespace */
+  namespace?: string;
+  cluster?: string;
+  /** Object list clusters */
+  clusters?: string[];
+  queryParams?: QueryParameters;
+}): [Array<InstanceType<T>> | null, ApiError | null] &
+  QueryListResponse<KubeList<InstanceType<T>>, InstanceType<T>, ApiError> {
+  if (clusters && clusters.length > 0) {
+    return _useKubeObjectLists({
+      kubeObjectClass,
+      namespace,
+      clusters: clusters,
+      queryParams,
+    });
+  } else {
+    return _useKubeObjectList<T>({
+      kubeObjectClass,
+      namespace,
+      cluster: cluster,
+      queryParams,
+    });
+  }
+}
+
+/**
+ * Returns a combined list of Kubernetes objects and watches for changes from the clusters given.
+ *
+ * @private please use useKubeObjectList
+ */
+function _useKubeObjectLists<T extends KubeObjectClass>({
+  kubeObjectClass,
+  namespace,
+  clusters,
+  queryParams,
+}: {
+  /** Class to instantiate the object with */
+  kubeObjectClass: T;
+  /** Object list namespace */
+  namespace?: string;
+  /** Object list clusters */
+  clusters: string[];
+  queryParams?: QueryParameters;
+}): [Array<InstanceType<T>> | null, ApiError | null] &
+  QueryListResponse<KubeList<InstanceType<T>>, InstanceType<T>, ApiError> {
+  const clusterResults: Record<string, ReturnType<typeof useKubeObjectList>> = {};
+
+  for (const cluster of clusters) {
+    clusterResults[cluster] = useKubeObjectList({
```

Comment:
Having hooks in a loop is a bit sketchy (although we already have loops like that in just a couple of places where we also ensure that the parent component will remount) the useList is used pretty much everywhere. Ideally we'd use useQueries to fetch multiple queries at once but I think this is okay for now

Reply:
When clusters change top level components remount. Which means the order is fixed.